pennylane/pauli/dla/lie_closure.py

```diff
@@ -13,14 +13,147 @@
 # limitations under the License.
 """A function to compute the Lie closure of a set of operators"""
 # pylint: disable=too-many-arguments
+import itertools
 from functools import reduce
 
+from typing import Union, Iterable
 from copy import copy
 import numpy as np
 
 import pennylane as qml
 
 from pennylane.operation import Operator
+from ..pauli_arithmetic import PauliWord, PauliSentence
+
+
+def lie_closure(
+    generators: Iterable[Union[PauliWord, PauliSentence, Operator]],
+    max_iterations: int = 10000,
+    verbose: int = 0,
+    pauli: bool = False,
+) -> Iterable[Union[PauliWord, PauliSentence, Operator]]:
+    r"""Compute the dynamical Lie algebra from a set of generators.
+
+    The Lie closure, pronounced "Lee" closure, is a way to compute the so-called dynamical Lie algebra (DLA) of a set of generators :math:`\mathcal{G} = \{G_1, .. , G_N\}`.
+    For such generators, one computes all nested commutators :math:`[G_i, [G_j, .., [G_k, G_\ell]]]` until no new operators are generated from commutation.
+    All these operators together form the DLA, see e.g. section IIB of `arXiv:2308.01432 <https://arxiv.org/abs/2308.01432>`__.
+
+    Args:
+        generators (Iterable[Union[PauliWord, PauliSentence, Operator]]): generating set for which to compute the
+            Lie closure.
+        max_iterations (int): maximum depth of nested commutators to consider. Default is ``10000``.
+        verbose (Union[int, bool]): verbosity during Lie closure calculation. Default is ``0``.
```

Comment:
Maybe just a bool input makes sense here?

Reply:
Since @dwierichs made the same point I am gonna change it to that :)